presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -246,25 +246,29 @@ public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, Con
     @Override
     public boolean supportsMissingColumnsOnInsert()
     {
-        return false;
+        return true;
     }
 
     @Override
     public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle, List<ColumnHandle> columns)
     {
         JdbcTableHandle handle = (JdbcTableHandle) tableHandle;
-        List<JdbcColumnHandle> allColumns = phoenixClient.getColumns(session, handle);
-        List<JdbcColumnHandle> nonRowkeyColumns = allColumns.stream()
-                .filter(column -> !ROWKEY.equalsIgnoreCase(column.getColumnName()))
+        Optional<String> rowkeyColumn = phoenixClient.getColumns(session, handle).stream()
+                .filter(column -> ROWKEY.equalsIgnoreCase(column.getColumnName()))
+                .map(JdbcColumnHandle::getColumnName)
```

Comment:
```suggestion
                .map(JdbcColumnHandle::getColumnName)
                .filter(ROWKEY::equalsIgnoreCase)
```

What if we could switch two lines, would it be better ?

Reply:
Yep. Much nicer. (A bit embarrassing that I missed that)